### PR TITLE
feature: S3C-4173-listAccounts-withFilters

### PR DESCRIPTION
**Pull request by anurag4DSB**
Filters with `customAttributes` for listAccounts.

Arsenal PR: https://github.com/scality/Arsenal/pull/1366
Vault PR: https://github.com/scality/Vault/pull/1500
Metadata PR: https://github.com/scality/MetaData/pull/1936

### CODE REVIEW

File: lib/IAMClient.js [hunk 1]
```diff
@@ -335,23 +335,42 @@ class VaultClient {
...
+                'filterKey and filterKeyStartsWith parameters are exclusive');
+        }
+
+        if (filterKey && filterKey.length !== 0) {
+            assert(typeof filterKey === 'string', 'filter should be a string');
+            data.filterKey = filterKey;
+        }
+        if (filterKeyStartsWith && filterKeyStartsWith.length !== 0) {
+            assert(typeof filterKeyStartsWith === 'string',
+                'filter should be a string');
```
Comment (saitharunthodupunuru) [inline lib/IAMClient.js]: Error message should be `filterKeyStartsWith` should be a string

File: lib/IAMClient.js [hunk 2]
```diff
@@ -335,23 +335,42 @@ class VaultClient {
...
-            data.MaxItems = options.maxItems;
+            data.MaxItems = maxItems;
+        }
+        if (filterKey && filterKeyStartsWith) {
+            assert(false,
+                'filterKey and filterKeyStartsWith parameters are exclusive');
+        }
+
+        if (filterKey && filterKey.length !== 0) {
+            assert(typeof filterKey === 'string', 'filter should be a string');
```
Comment (saitharunthodupunuru) [inline lib/IAMClient.js]: Error message should be `filterKey` should be a string

**Review by saitharunthodupunuru — CHANGES_REQUESTED** [review]

File: lib/IAMClient.js (hunk 1 above)
Comment (anurag4DSB) [reply lib/IAMClient.js]: Changed in review commit: [`74a60d2` (#291)](https://github.com/scality/vaultclient/pull/291/commits/74a60d2633354ec5ce14c10e1d5644d0e3956d34)

File: lib/IAMClient.js (hunk 2 above)
Comment (anurag4DSB) [reply lib/IAMClient.js]: Changed in review commit: [`74a60d2` (#291)](https://github.com/scality/vaultclient/pull/291/commits/74a60d2633354ec5ce14c10e1d5644d0e3956d34)

**Review by saitharunthodupunuru — APPROVED** [review]
LGTM

**Comment by anurag4DSB** [comment]
Tests failing due to CI.

**Review by vrancurel — APPROVED** [review]

**Comment by anurag4DSB** [comment]
ping

**Pull request merged**